pytorch_lightning/trainer/trainer.py

```diff
@@ -412,6 +412,46 @@ def __init__(
         # Callback system
         self.on_init_end()
 
+    def setup_trainer(self, model: LightningModule):
```

Comment:
I think this refactor broke `trainer.test(resume_from_checkpoint="some_path", ...)`
I don't see where `self.trainer.checkpoint_connector.restore_weights()` is called in the evaluation loop or in the trainer. before this happened as part of the TrainLoop's `setup_training`

In `Trainer`'s `run_evaluation` I don't see any calls to the checkpoint connector either.

Reply:
this is done in `setup_training` here: https://github.com/PyTorchLightning/pytorch-lightning/pull/5388/files#diff-6b21474ed45079f01dfa45ee3b9d40d23efe693c9c005ce897e25384ef425349

as of now restore is done while calling `.fit`. Previously it was happening during `.test()` too and was reloading the `resume_from_checkpoint` state and model weights that were wrong. But for your use-case, I opened an issue for reloading the callback states optionally.